x-pack/plugins/ml/public/application/datavisualizer/index_based/index_data_visualizer.tsx

```diff
@@ -180,19 +181,51 @@ export const IndexDataVisualizerPage: FC = () => {
     // eslint-disable-next-line react-hooks/exhaustive-deps
     [mlLocator, mlFeaturesDisabled]
   );
+  const { euiTheme } = useEuiTheme();
+
   return IndexDataVisualizer ? (
     <Fragment>
       {IndexDataVisualizer !== null ? (
         <>
           <MlPageHeader>
-            <FormattedMessage
-              id="xpack.ml.dataVisualizer.pageHeader"
-              defaultMessage="Data Visualizer"
-            />
+            <EuiFlexGroup gutterSize="s" alignItems="center" direction="row">
+              <FormattedMessage
+                id="xpack.ml.dataVisualizer.pageHeader"
+                defaultMessage="Data Visualizer"
+              />
+              {esql ? (
+                <>
+                  <EuiFlexItem grow={false}>
+                    <FormattedMessage id="xpack.ml.datavisualizer" defaultMessage="(ES|QL)" />
+                  </EuiFlexItem>
+                  <EuiFlexItem grow={false} css={{ marginTop: euiTheme.size.xs }}>
+                    <EuiBetaBadge
+                      label={i18n.translate(
+                        'xpack.ml.datavisualizer.selector.technicalPreviewBadge.label',
+                        {
+                          defaultMessage: 'Technical preview',
+                        }
+                      )}
+                      size="m"
+                      color="hollow"
+                      tooltipContent={i18n.translate(
+                        'xpack.ml.datavisualizer.selector.technicalPreviewBadge.tooltipMsg',
+                        {
+                          defaultMessage:
+                            'This functionality is in technical preview and may be changed or removed completely in a future release. Elastic will work to fix any issues, but features in technical preview are not subject to the support SLA of official GA features.',
+                        }
+                      )}
+                      tooltipPosition={'right'}
+                    />
```

Comment:
Could you please replace with https://github.com/elastic/kibana/blob/0997cb25238f812a3eff67733bb16ef1521abea4/x-pack/plugins/ml/public/application/components/technical_preview_badge/technical_preview_badge.tsx#L14